File: 20-valid-parentheses/20-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const parens = {'(':')', '[':']', '{':'}'};
    let ans = true;
    let chars = [];
    s.split('').forEach((c)=>{
        if(parens[c]){
            chars.push(c);
        } else {
            if(parens[chars[chars.length-1]] === c){
                chars.pop();
            } else {
                ans = false;
            }
        }
        
    })
    return ans && chars.length === 0;
};

File: 20-valid-parentheses/20-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        dictionary = {'(':')','{':'}','[':']'}
        stack = []
        for c in s:
            if c in dictionary:
                stack.append(c)
            elif len(stack)>0 and c == dictionary[stack[-1]]:
                stack.pop()
            else:
                return False
        return not stack
        